server/api/delete-account.js

```diff
@@ -0,0 +1,73 @@
+const { getSdk, getTrustedSdk, handleError } = require('../api-util/sdk');
+
+module.exports = (req, res) => {
+  const { currentPassword } = req.body;
+
+  const sdk = getSdk(req, res);
+
+  const stripeRelatedStatesForBookings = [
+    'state/pending-payment',
+    'state/preauthorized',
+    'state/accepted',
+    'state/delivered',
+  ];
+  const stripeRelatedStatesForPurchases = [
+    'state/pending-payment',
+    'state/purchased',
+    'state/delivered',
+    'state/disputed',
+    'state/received',
+  ];
+  const stripeRelatedStatesForNegotiation = [
+    'state/pending-payment',
+    'state/offer-accepted',
+    'state/delivered',
+    'state/changes-requested',
+    'state/completed',
+  ];
```

Comment:
Perhaps these could be moved to the file-scope
(above this middleware definition `module.exports = (req, res) => {`)

That way it would be easier to follow what logic is actually inside it.